types/handlebars/index.d.ts

```diff
@@ -52,7 +62,7 @@ declare namespace Handlebars {
     export function blockParams(obj: any[], ids: any[]): any[];
     export function Exception(message: string): void;
     export function log(level: number, obj: any): void;
-    export function parse(input: string): hbs.AST.Program;
+    export function parse(input: string, options?: ParseOptions): hbs.AST.Program;
```

Comment:
Can you provide some links to docs/discussion, indicating that these options are not only possible to pass to `parse()`, but are also **public API**?

Looking at https://handlebarsjs.com/reference.html I see no reference to the `parse()` function at all, let alone options.

Reply:
It's not documented on the website, but in [docs directory of github](https://github.com/wycats/handlebars.js/blob/master/docs/compiler-api.md#ast), as it's for advanced usage to manipulate output AST before rendering the template.

The options parameter is not documented there sadly, but it seems possible to pass option to the precompile method instead. I'll make some tests and will let you known.

Reply:
Well ... This options parameter on `parsed` function seems to be the only way to pass some options (`ignoreStandalone` and `srcName`), passing them later in the workflow  (precompile or compile) seems has no effect.

Reply:
> The options parameter is not documented there sadly, but it seems possible to pass option to the precompile method instead. I'll make some tests and will let you known.


Because DT types are supposed to be strictly for the public API of the packages they describe, we need to get clarity around what the public API is before merging in your changes (i.e., the options parameter is public, and the properties on the options object are also public in the SemVer major version sense).

Any written statement would be fine, like...
* a github issue asking whether this is part of the public API, with a response indicating it is
* a merged/approved PR modifying the docs to make it clear that this is public

Reply:
Issue [opened here](https://github.com/wycats/handlebars.js/issues/1480), but still no answer.

Reply:
As soon as we can confirm that this is really Public API, I'm happy to review/approve this change promptly.